tests/ui/derivable_impls.rs

```diff
@@ -0,0 +1,121 @@
+use std::collections::HashMap;
+
+struct FooDefault {
+    a: bool,
+    b: i32,
+    c: u64,
+    d: Vec<i32>,
+    e: FooND1,
+    f: FooND2,
+    g: HashMap<i32, i32>,
+    h: (i32, Vec<i32>),
+    i: [Vec<i32>; 3],
+    j: [i32; 5],
+    k: Option<i32>,
+}
```

Comment:
How about a slice?

Reply:
My current `is_default_equivalent` doesn't detect slice. Is it generally true that `&(T::default())` is equal to `(&T)::default()`?

Reply:
I don't think so. But you can see the Default impl [in the docs] (https://doc.rust-lang.org/std/primitive.slice.html#impl-Default).